src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -1085,6 +1096,18 @@ private async Task<Task> SetActiveTreeAnalysisTaskAsync(Func<Tuple<Task, Cancell
             }
         }
 
+        private async Task WaitForExecutingTaskAsync(Task executingTask)
```

Comment:
📝 This would benefit from something like [`NoThrowAwaitable`](https://github.com/microsoft/vs-threading/blob/558f24c576cc620a00b20ed1fa90a5e2d13b0440/src/Microsoft.VisualStudio.Threading/TplExtensions.cs#L242-L252)